### ISSUE TITLE
How to create custom function with 2 arguments?

### ISSUE THREAD

**Issue by S4TNEW** [stephencelis/SQLite.swift#469]
I created a custom function that takes 2 arguments:

```
let s4tEqualsIgnoreCase: (Expression<String>, Expression<String>) -> Expression<Bool?> = (
    try! db.createFunction("s4tEqualsIgnoreCase")
    { value, value2 in
        return value.lowercaseString == value2.lowercaseString
    }
)
```

Usage:

```
let condition = s4tEqualsIgnoreCase(FLFCustomer.kName, Expression<String>(name))
```

It compiles but when I run the code I get an error:

> fatal error: 'try!' expression unexpectedly raised an error: wrong number of arguments to function s4tEqualsWithCase()

Taking a look into the definition of 'createFunction()' in CustomFunctions.swift I can see that the number of arguments is hard coded as '1'.

So my questions is: How can I define and use a custom function that takes 2 arguments?

**Comment by S4TNEW** [comment]
Could that be the reason?

In CustomFunctions.swift the functions taking two arguments call createFunction() that takes the argumentCount, but it will always be called with argumentCount = 1.

So I think the following implementation
```
public func createFunction<Z : Value, A : Value, B : Value>(function: String, deterministic: Bool = false, _ block: (A, B) -> Z) throws -> (Expression<A>, Expression<B>) -> Expression<Z> {
    let fn = try createFunction(function, 1, deterministic) { args in block(value(args[0]), value(args[1])) }
    return { a, b in fn([a, b]) }
}
```

should read
```
[...]
let fn = try createFunction(function, 2, deterministic) { args in block(value(args[0]), value(args[1])) }
[...]
```

Any thoughts on this?

**Comment by jberkel** [comment]
#481

**Comment by jberkel** [comment]
Closed as dup of #481
